src/main/java/com/odysseusinc/arachne/executionengine/model/descriptor/r/RExecutionRuntime.java

```diff
@@ -70,6 +72,8 @@ public static RExecutionRuntime fromREnvLock(REnvLock rEnvLock) {
                     return dependency;
                 })
                 .collect(Collectors.toList());
+        
+        runtime.dependencies = (RDependency[])rDependencies.toArray();
```

Comment:
This should be changed to `runtime.dependencies = rDependencies.toArray(new RDependency[0]);` And class casting is not needed